docs/whichkey/reference.md

```diff
@@ -0,0 +1,281 @@
+---
+id: reference
+title: Reference
+---
+Which-key extension aims to provides similar keybinding popup using the [QuickPick](https://code.visualstudio.com/api/references/vscode-api#QuickPick)
+API from vscode.
+This extension was broken out of `VSpaceCode`, which was created to emulate the keybinding of spacemacs in VSCode ([VSpaceCode](./../installation.md)),
+to decouple the core and the definition of bindings.
+
+## Commands
+
+Commands are the main way to inference with which-key.
+
+### Show
+
+Command: `whichkey.show`
+
+Args: `string`, `Array<BindingItem>`, `undefined`
+
+Description:
+A command to show which-key menu.
+
+- `string` is the key of the [registered](#register) bindings that will be use to display a which-key menu.
+- `Array<BindingItem>` is an array of [`BindingItem`](#bindingitem) that will be used directly to display a which-key menu.
+- `undefined` means the default bindings defined at `whichkey.bindings` in the configuration will be used to display a which-key menu.
+
+### Register
+
+Command: `whichkey.register`
+
+Args:
+
+```ts
+{
+    bindings: string;
+    overrides?: string;
+    title?: string;
+}
+```
+
+Description:
+A command to register config which-key so a canonical record of the bindings lives in memory.
+
+- `bindings` is the configuration location of the bindings. e.g. `vspacecode.bindings`.
+- `overrides` is the configuration location of the overrides. e.g. `vspacecode.overrides`.
+- `title` is the title of the which-key menu to display on the top-level.
+
+### Trigger Key
+
+Command: `whichkey.triggerKey`
+
+Args: `string`, `{ key: string, when?: string }`
+
+Description:
+A command used primarily for accepting non-characters key in which-key menu and pass in `when` clause via vscode shortcut.
+
+:::note
+
+#### Non-character key
+
+The which-key menu relies QuickPick's input. Because it is a text input, we can not capture non-character keys like arrow keys, tab key, and keys with
+modifier like C-x. We can capture those keys using vscode shortcut with `whichkeyVisible` as when clause of the shortcut.
+See [Extra](extra#use-non-character-keys) page for detail on usage.
+:::
+
+:::note
+
+#### Pass-in when clause
+
+Since vscode doesn't allows extension to read context key-value ([vscode#10471](https://github.com/microsoft/vscode/issues/10471)),
```

Comment:
```suggestion
Since vscode doesn't allow extension to read context key-value ([vscode#10471](https://github.com/microsoft/vscode/issues/10471)),
```